# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/gccx/install_so.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./install-sh.orig	2003-07-20 11:40:28.000000000 +0200
+++ ./install-sh	2003-11-02 17:04:13.000000000 +0100
@@ -283,7 +283,13 @@
 
 # Now rename the file to the real destination.
 
-	$doit $mvcmd "$dsttmp" "$dstdir/$dstfile"
+	if echo $dstdir/$dstfile | grep -q 'lib/.*\.so' ; then
+		$doit $mvcmd $dsttmp $dstdir/$dstfile
+	else
+		$doit $rmcmd -f $dstdir/$dstfile &&
+		$doit $mvcmd $dsttmp $dstdir/$dstfile
+	fi
+ 
 
 fi &&
 
